docker/all-in-one/entrypoint.sh

```diff
@@ -50,14 +54,40 @@ function enable_swap {
   swapon /mnt/swapfile
 }
 
-function create_lsn_checkpoint_file {
-  if [ ! -f "${DATA_VOLUME_MOUNTPOINT}/latest-lsn-checkpoint" ]; then
-    echo "0/0" > "${DATA_VOLUME_MOUNTPOINT}/latest-lsn-checkpoint"
-    chown postgres:postgres "${DATA_VOLUME_MOUNTPOINT}/latest-lsn-checkpoint"
-    chmod 0300 "${DATA_VOLUME_MOUNTPOINT}/latest-lsn-checkpoint"
-  fi
+function push_lsn_checkpoint_file {
+    if [ "${PLATFORM_DEPLOYMENT:-}" != "true" ]; then
+      echo "Skipping push of LSN checkpoint file"
+      return
+    fi
+
+    /usr/bin/admin-mgr lsn-checkpoint-push --immediately || echo "Failed to push LSN checkpoint"
 }
 
+function graceful_shutdown {
+  echo "$(date): Received SIGINT. Shutting down."
+  supervisorctl stop postgresql
+
+  # Postgres ships the latest WAL file using archive_command during shutdown, in a blocking operation
+  # This is to ensure that the WAL file is shipped, just in case
+  sleep 0.2
+  push_lsn_checkpoint_file
+
+  kill -s TERM "$(supervisorctl pid)"
+}
+
+function enable_autoshutdown {
+    sed -i "s/autostart=.*/autostart=true/" /etc/supervisor/db-only/supa-shutdown.conf
+}
+
+function enable_lsn_checkpoint_push {
+    sed -i "s/autostart=.*/autostart=true/" /etc/supervisor/db-only/lsn-checkpoint-push.conf
```

Comment:
Isn't this required even if not on a box configured as `db-only`? 🤔 

Similar thought for the remaining `supa-shutdown` and `lsn-checkpoint-push` files for supervisord or references, that bring up `db-only`.

Reply:
Ah, the `db-only` folder is the default (and minimal) set of services which supervisord runs:
https://github.com/supabase/postgres/blob/develop/docker/all-in-one/etc/supervisor/supervisord.conf#L169-L170

During regular operation if `POSTGRES_ONLY` is not true, `services/*conf` also gets added:
https://github.com/supabase/postgres/blob/develop/docker/all-in-one/entrypoint.sh#L234-L238

Though, good point - this causes quite a bit of confusion - better to rename the `db-only` folder to `default` or `base-services`

Reply:
Got it. Thanks for the points it might have escaped me but, yeah, unless you are very comfortable already with the term `db-only` it might be a bit misleading. Change to `base-services`, since the other folder is already called `services` sounds good to me 👌 

Regarding checks/tests:
- `Testinfra Integration Tests / test-all-in-one ` is failing with
```
FAILED testinfra/test_all_in_one.py::test_postgrest_responds_to_requests - requests.exceptions.ConnectionError:
HTTPConnectionPool(host='localhost', port=8000): Max retries exceeded with url:
 /rest/v1/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection 
object at 0x7f8a8357bf70>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

- `Testinfra Integration Tests / test-ami` is failing with 
```
Build 'amazon-ebssurrogate.source' errored after 11 minutes 52 seconds: 
Script exited with non-zero exit status: 2.Allowed exit codes are: [0]
```